src/build/emitter.ts

```diff
@@ -826,6 +826,14 @@ export function emitWebIdl(
     entity: { comment?: string; deprecated?: boolean },
     print: (s: string) => void
   ) {
+    if (
+      entity.deprecated &&
+      entity.comment?.endsWith(`*/`) &&
+      !entity.comment?.includes("@deprecated")
+    ) {
+      entity.comment = entity.comment.replace("*/", "* @deprecated\n */");
+    }
+
     if (entity.comment) {
```

Comment:
Instead of a new, side-effecting section, I'd rather see the existing code refactored into exclusive comment/no-comment sections:

```ts
if (entity.comment) {
  // previous code, plus code to insert `@deprecated`
}
else if (entity.deprecated) {
  // previous code
  print("/** @deprecated")
}

Reply:
Done in c5b68a87d5bebdb19061dc78f11dff05dd95e573. A few edge cases where conversion from single to multi-line comment was required are now also handled.